// antd-style/es/containers/AppContainer/AppContainer.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["children", "appearance", "themeMode", "customToken", "customStylish", "globalStyle"];
import { memo } from 'react';
import { ThemeModeContext } from "../../context";
import { ThemeProvider } from "../ThemeProvider";
import { AntdProvider } from "./AntdProvider";
import { GlobalStyle } from "./GloabalStyle";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export var AppContainer = /*#__PURE__*/memo(function (_ref) {
  var children = _ref.children,
      appearance = _ref.appearance,
      themeMode = _ref.themeMode,
      customToken = _ref.customToken,
      customStylish = _ref.customStylish,
      globalStyle = _ref.globalStyle,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/_jsx(ThemeModeContext.Provider, {
    value: {
      themeMode: themeMode || 'light',
      appearance: appearance || 'light'
    },
    children: /*#__PURE__*/_jsx(AntdProvider, _objectSpread(_objectSpread({}, props), {}, {
      children: /*#__PURE__*/_jsxs(ThemeProvider, {
        customToken: customToken,
        customStylish: customStylish,
        children: [/*#__PURE__*/_jsx(GlobalStyle, {
          globalStyle: globalStyle
        }), children]
      })
    }))
  });
});
