// antd-style/lib/containers/AppContainer/AppContainer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AppContainer = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = require("react");

var _context = require("../../context");

var _ThemeProvider = require("../ThemeProvider");

var _AntdProvider = require("./AntdProvider");

var _GloabalStyle = require("./GloabalStyle");

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["children", "appearance", "themeMode", "customToken", "customStylish", "globalStyle"];
var AppContainer = /*#__PURE__*/(0, _react.memo)(function (_ref) {
  var children = _ref.children,
      appearance = _ref.appearance,
      themeMode = _ref.themeMode,
      customToken = _ref.customToken,
      customStylish = _ref.customStylish,
      globalStyle = _ref.globalStyle,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_context.ThemeModeContext.Provider, {
    value: {
      themeMode: themeMode || 'light',
      appearance: appearance || 'light'
    },
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_AntdProvider.AntdProvider, (0, _objectSpread2.default)((0, _objectSpread2.default)({}, props), {}, {
      children: /*#__PURE__*/(0, _jsxRuntime.jsxs)(_ThemeProvider.ThemeProvider, {
        customToken: customToken,
        customStylish: customStylish,
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_GloabalStyle.GlobalStyle, {
          globalStyle: globalStyle
        }), children]
      })
    }))
  });
});
exports.AppContainer = AppContainer;
