// antd-style/lib/stylish/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useInternalStylish = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _useAntdToken = require("../hooks/useAntdToken");

var _css = require("@emotion/css");

var _polished = require("polished");

var _react = require("react");

var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7, _templateObject8, _templateObject9, _templateObject10, _templateObject11;

/**
 * 一组统一封装好的 antd 标准样式
 */
var useInternalStylish = function useInternalStylish() {
  var token = (0, _useAntdToken.useAntdToken)();
  return (0, _react.useMemo)(function () {
    var containerBgHover = (0, _css.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n      cursor: pointer;\n      transition: 150ms background-color ease-in-out;\n      &:hover {\n        background: ", ";\n      }\n    "])), token.colorFillQuaternary);
    var dataInputContainerHover = (0, _css.css)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2.default)(["\n      color: ", ";\n      background-color: ", ";\n      border-color: transparent;\n    "])), token.colorText, token.colorFillTertiary);
    var dataInputContainerFocused = (0, _css.css)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2.default)(["\n      color: ", " !important;\n      background-color: ", " !important;\n      border-color: ", " !important;\n      box-shadow: none;\n    "])), token.colorText, token.colorFillQuaternary, token.colorPrimary);
    var defaultButtonBase = (0, _css.css)(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2.default)(["\n      color: ", ";\n      background: ", ";\n      border-color: transparent;\n    "])), token.colorTextSecondary, token.colorFillQuaternary);
    return {
      defaultButton: (0, _css.css)(_templateObject5 || (_templateObject5 = (0, _taggedTemplateLiteral2.default)(["\n        ", ";\n\n        &:hover {\n          color: ", " !important;\n          background: ", " !important;\n          border-color: transparent !important;\n        }\n        &:focus {\n          ", ";\n          border-color: ", " !important;\n        }\n      "])), defaultButtonBase, token.colorText, token.colorFillSecondary, defaultButtonBase, token.colorPrimary),
      textInfo: (0, _css.css)(_templateObject6 || (_templateObject6 = (0, _taggedTemplateLiteral2.default)(["\n        color: ", ";\n        &:hover {\n          color: ", ";\n        }\n      "])), token.colorTextSecondary, token.colorText),
      textDefault: (0, _css.css)(_templateObject7 || (_templateObject7 = (0, _taggedTemplateLiteral2.default)(["\n        color: ", ";\n      "])), token.colorTextSecondary),
      containerBgHover: (0, _css.css)(_templateObject8 || (_templateObject8 = (0, _taggedTemplateLiteral2.default)(["\n        cursor: pointer;\n        transition: 150ms background-color ease-in-out;\n\n        &:hover {\n          background: ", ";\n        }\n      "])), token.colorFillQuaternary),
      containerBgL2: (0, _css.css)(_templateObject9 || (_templateObject9 = (0, _taggedTemplateLiteral2.default)(["\n        ", ";\n        border-radius: 4px;\n        background: ", ";\n\n        &:hover {\n          background: ", ";\n        }\n      "])), containerBgHover, token.colorFillQuaternary, token.colorFillTertiary),
      dataInputContainerFocused: dataInputContainerFocused,
      dataInputContainer: (0, _css.css)(_templateObject10 || (_templateObject10 = (0, _taggedTemplateLiteral2.default)(["\n        &:hover {\n          ", "\n        }\n        &:focus {\n          ", "\n        }\n      "])), dataInputContainerHover, dataInputContainerFocused),
      backgroundBlur: (0, _css.css)(_templateObject11 || (_templateObject11 = (0, _taggedTemplateLiteral2.default)(["\n        background: ", ";\n        backdrop-filter: blur(10px);\n      "])), (0, _polished.transparentize)(0.4)(token.colorBgElevated))
    };
  }, [token]);
};

exports.useInternalStylish = useInternalStylish;
